src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php

```diff
@@ -341,6 +341,39 @@ public function iShouldSeeThatTheVariantHasZeroOnHandQuantity(ProductVariantInte
     }
 
     /**
+     * @Then /^(\d+) units of (this product) should be on hold$/
+     */
+    public function unitsOfThisProductShouldBeOnHold($quantity, ProductInterface $product)
+    {
+        Assert::true(
+            $this->indexPage->hasOnHoldQuantity($product->getFirstVariant(), $quantity),
+            sprintf('This "%s" variant should has "%s" quantity, but it does not.', $product->getFirstVariant()->getName(), $quantity)
+        );
+    }
+
+    /**
+     * @Then /^(\d+) units of (this product) should be on hand$/
+     */
+    public function unitsOfThisProductShouldBeOnHand($quantity, ProductInterface $product)
+    {
+        Assert::true(
+            $this->indexPage->hasOnHandQuantity($product->getFirstVariant(), $quantity),
+            sprintf('This "%s" variant should has "%s" quantity, but it does not.', $product->getFirstVariant()->getName(), $quantity)
+        );
+    }
+
+    /**
+     * @Then /^there should be no units of (this product) on hold$/
+     */
+    public function thereShouldBeNoUnitsOfThisProductOnHold(ProductInterface $product)
+    {
+        Assert::false(
+            $this->indexPage->hasOnHoldQuantity($product->getFirstVariant(), 1),
```

Comment:
Why checking for `1`? If `0` is not displayed we ca simply check if there is `onHold` label or not.